google/cloud/bigtable/row_data.py

```diff
@@ -627,6 +628,7 @@ def build_updated_request(self):
         r_kwargs = {
             "table_name": self.message.table_name,
             "filter": self.message.filter,
+            "app_profile_id": self.message.app_profile_id,
```

Comment:
I think this works, but it puts us in the same position as before: if ReadRowsRequest ever adds a new field, we will end up dropping that field on retry. I think one of 2 things should happen:
1. clone the entry request and overwrite the RowSet & limit. This way if new fields are added, we are sure to copy them
2. Add a unit test that ensure no new fields are added with a comment to instructing someone to update this code

I think 1 is definitely preferable